### PR TITLE
Craft cli integration

### PR DESCRIPTION
**Pull request by facundobatista**

### CODE REVIEW

File: charmcraft/commands/build.py [hunk 1]
```diff
@@ -443,20 +432,24 @@ def pack_charm_in_instance(
...
             build_on_index=build_on_index,
         ) as instance:
+            emit.progress("Packing the charm")
             try:
-                instance.execute_run(
-                    cmd,
-                    check=True,
-                    cwd=instance_output_dir,
-                )
+                with emit.open_stream(f"Running {cmd}") as stream:
```
Comment (facundobatista) [inline charmcraft/commands/build.py:445]: x

File: charmcraft/logsetup.py [hunk 2]
```diff
@@ -141,4 +141,4 @@ def ended_crash(self, err):
             self._file_logger.exception("")
 
 
-message_handler = _MessageHandler()
+# message_handler = _MessageHandler()
```
Comment (facundobatista) [inline charmcraft/logsetup.py:144]: x

**Review by cmatsuoka — APPROVED** [review]
Output migration to emitter seems ok, and I see that there will be a follow-up for further adjustments. Also it seems that you're using the predictable log filename in both instances instead of only in the inner instance. Please verify if this is indeed what you want to do -- if not, this can be adjusted in a forthcoming PR.

File: charmcraft/charm_builder.py [hunk 3]
```diff
@@ -343,9 +339,8 @@ def main():
     """Run the command-line interface."""
     options = _parse_arguments()
 
-    logging.basicConfig(level=logging.DEBUG, format="%(message)s")
-
-    logger.debug("Starting charm builder")
+    logpath = get_managed_environment_log_path()
+    emit.init(EmitterMode.TRACE, "charm-builder", "Starting charm builder", log_filepath=logpath)
```
[the remark this answers is not in the document]
Comment (cmatsuoka) [reply charmcraft/charm_builder.py:343]: Of course not, this is only for the charm builder process that only runs inside.

File: charmcraft/commands/build.py [hunk 4]
```diff
@@ -443,20 +432,24 @@ def pack_charm_in_instance(
...
+                    instance.execute_run(
```
[the remark this answers is not in the document]
Comment (facundobatista) [reply charmcraft/commands/build.py:446]: We discussed a little about this, Claudio will open an issue in craft-cli for this functionality.

[the remark this answers is not in the document]
Comment (cmatsuoka) [reply charmcraft/commands/build.py:446]: Opened issue #582 with this idea.

**Pull request merged**